src/kaleidoscope/plugin/OneShot.cpp

```diff
@@ -22,266 +22,339 @@
 namespace kaleidoscope {
 namespace plugin {
 
-// ---- state ---------
+// ----------------------------------------------------------------------------
+// Configuration variables
 
-uint16_t OneShot::start_time_ = 0;
 uint16_t OneShot::time_out = 2500;
 uint16_t OneShot::hold_time_out = 250;
 int16_t OneShot::double_tap_time_out = -1;
-OneShot::key_state_t OneShot::state_[OneShot::ONESHOT_KEY_COUNT];
-Key OneShot::prev_key_;
-bool OneShot::should_cancel_ = false;
-bool OneShot::should_cancel_stickies_ = false;
 
-bool OneShot::isPressed() {
-  for (uint8_t i = 0; i < ONESHOT_KEY_COUNT; i++) {
-    if (state_[i].pressed)
+// ----------------------------------------------------------------------------
+// State variables
+
+OneShot::OneShotKeyState OneShot::state_[OneShot::ONESHOT_KEY_COUNT];
+uint16_t OneShot::start_time_ = 0;
+KeyAddr OneShot::prev_key_addr_ = UnknownKeyswitchLocation;
+uint8_t OneShot::release_countdown_ = 0;
+
+
+// ============================================================================
+// Public interface
+
+// ----------------------------------------------------------------------------
+// Configuration functions
+
+void OneShot::enableStickability(Key key) {
+  if (key >= ranges::OS_FIRST && key <= ranges::OS_LAST)
+    state_[key.getRaw() - ranges::OS_FIRST].stickable = true;
+}
+
+void OneShot::disableStickability(Key key) {
+  if (key >= ranges::OS_FIRST && key <= ranges::OS_LAST)
+    state_[key.getRaw() - ranges::OS_FIRST].stickable = false;
+}
+
+void OneShot::enableStickabilityForModifiers() {
+  for (uint8_t i{0}; i < ONESHOT_MOD_COUNT; ++i) {
```

Comment:
I changed all the for loops to this style, using brace initialization and the prefix increment operator. I had already written several like that before I noticed that the existing code used a different style. Either of these elements can be changed back from `i{0}` to `i = 0` and `++i` to `i++` if preferred; the compiled binary should be the same either way.